crypto/provider_core.c

```diff
@@ -673,13 +730,47 @@ static int core_thread_start(const OSSL_PROVIDER *prov,
     return ossl_init_thread_start(prov->provctx, handfn);
 }
 
+/*
+ * The FIPS module inner provider doesn't implement these.  They aren't
+ * needed there, since the FIPS module upcalls are always the outer provider
+ * ones.
+ */
+#ifndef FIPS_MODE
+static void core_put_error(const OSSL_PROVIDER *prov,
+                           uint32_t reason, const char *file, int line)
+{
+    /*
+     * If the uppermost 8 bits are non-zero, it's an OpenSSL library
+     * error and will be treated as such.  Otherwise, it's a new style
+     * provider error and will be treated as such.
+     */
+    if ((reason & 0xFF000000U) != 0) {
```

Comment:
should use the ```ERR_get_lib``` macro and have a define for zero?
This comment should be in the err.h file I think.

Reply:
You're right that there is `ERR_GET_LIB()`.  I'll use that.  A macro for zero becomes silly, though...

Reply:
Why? It's a magic number with special meaning.